adapters/httpapi_curl.c

```diff
@@ -401,6 +401,31 @@ HTTPAPI_RESULT HTTPAPI_ExecuteRequest(HTTP_HANDLE handle, HTTPAPI_REQUEST_TYPE r
 
                     break;
 
+                case HTTPAPI_REQUEST_HEAD:
+                    if (curl_easy_setopt(httpHandleData->curl, CURLOPT_HTTPGET, 1L) != CURLE_OK)
```

Comment:
Do you need to explicitly set the CURLOPT_HTTPGET?  My read of https://curl.haxx.se/libcurl/c/CURLOPT_NOBODY.html is that the NOBODY option is enough.